### PR TITLE
Upgrade to API version 2

### PR DESCRIPTION
**Pull request by servingUpAces**
Upgrade to use version 2 of the notification APIs
- change to ResponseNotificationm response to .
- change to Notification, response to getNotificaitonById()
- change to NotificationList, response to getNotifications()
- New filters for getNotifications(status, notificationType, reference, olderThanId)
- Update the README to reflect all these changes.
Now at version 2.0.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+##2.0.0
+
+###Changed
+* Using version 2 of the notification-api.
```
Comment (leohemsted) [inline CHANGELOG.md]: need to include no longer being able to use separate `service_id` and `api_key_id` in this

File: bin/test_client.rb [hunk 2]
```diff
@@ -106,49 +108,54 @@ def field_should_be_nil(fields, obj, method_name)
...
      status
-     service
      sent_at
-     sent_by
      template
-     template_version
+     body
+     subject
      created_at
-     updated_at
```
Comment (leohemsted) [inline bin/test_client.rb]: why has `updated_at` been removed? do we no longer send that back?

File: bin/test_client.rb [hunk 3]
```diff
@@ -106,49 +108,54 @@ def field_should_be_nil(fields, obj, method_name)
...
-     body
-     notification_type
+     reference
+     email_address
+     type
      status
-     service
      sent_at
-     sent_by
      template
```
Comment (leohemsted) [inline bin/test_client.rb]: we should drill down into template and assert that the id, version and uri are set

File: notifications-ruby-client.gemspec [hunk 4]
```diff
@@ -11,7 +11,7 @@ Gem::Specification.new do |spec|
     "dazahern"
   ]
 
-  spec.email         = ["info@bitzesty.com", "daz.ahern@digital.cabinet-office.gov.uk"]
+  spec.email         = ["notify@digital.cabinet-office.gov.uk"]
```
Comment (leohemsted) [inline notifications-ruby-client.gemspec]: do we want to remove bitzesty/dazahern from the spec.authors above?

File: README.md [hunk 5]
```diff
@@ -28,43 +28,224 @@ Generate an API key by logging in to GOV.UK Notify [GOV.UK Notify](https://www.n
...
+   <tr>
+    <td>
+      <pre>429</pre>
+    </td>
+    <td>
+      <pre>
+      [{
+          "error": "TooManyRequestsError",
+          "message": "Exceeded send limits (50) for today"
+      }]
```
Comment (servingUpAces) [inline README.md]: That is the actual content of the message

File: bin/test_client.rb (hunk 2 above)
Comment (servingUpAces) [inline bin/test_client.rb]: No, but we send completed_at which is only present if the notification is delivered or failed.

File: notifications-ruby-client.gemspec (hunk 4 above)
Comment (servingUpAces) [inline notifications-ruby-client.gemspec]: We can't leave authors empty, should I put Notify as the author?

**Pull request merged**